### DISCUSSION TITLE
ST7796 on Mega2560

### DISCUSSION THREAD

**Discussion by RafaelxVS** [Bodmer/TFT_eSPI#2374] [Compatible displays and setup files]
Hi,
Please, I need some advice.
I'm using the display model ST7796 V1.0 ([link](http://www.lcdwiki.com/4.0inch_SPI_Module_ST7796)) and an Arduino Mega2560.
I unsoldered D1 and jumped the contacts like the instructions here: [https://github.com/Bodmer/TFT_eSPI/discussions/898](url)
I can't do the things work.
The connections:
TFT // Mega
VCC // 5V
GND // GND
CS // A5
Reset // A4
DC/CD // A3
SDI(MOSI) // 51
SCK // 52
LED // A0
SDO (MISO) // 50

Attached my User_Setup and User_Select.
[TFT_eSPI.zip](https://github.com/Bodmer/TFT_eSPI/files/10548543/TFT_eSPI.zip)

Please, can anyone help with this issue?

Thanks!
Regards.

**Comment by Bodmer** [comment]
The Mega uses 5v logic and thus cannot be directly connected to the display without a logic level converter. The simplest converter is a resistor divider as described in Step 1 here:
https://www.google.com/amp/s/www.instructables.com/Arduino-TFT-display-and-font-library/%3famp_page=true

This library is also more suited to higher performance 32 bit processors (such as the Pico RP2040) due to the memory resource demands. For that processor suitable connections are in this setup file:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup60_RP2040_ILI9341.h

**Reply by RafaelxVS** [reply]
Hi Bodmer,
Great, I'll make the voltage divisor.
Doing this, it'll be suitable on Mega2560?

Is there a better display (4") for Arduino Nano/Due/Mega? I don't need the touch screen.

I'll search about Pico RP2040.

Thanks!
Regards,
Rafael

**Reply by Bodmer** [reply]
There are some options that include 5v logic converters but it would be cheaper to get a low cost 32 bit processor and you would have the memory size and performance to use this library.

**Reply by RafaelxVS** [reply]
> There are some options that include 5v logic converters but it would be cheaper to get a low cost 32 bit processor and you would have the memory size and performance to use this library.

Hi,
What are the best options? You told about Pico RP2040. Only Raspberry has low cost options with 32bits?
Thanks again.
Regards,
Rafael

**Reply by Bodmer** [reply]
STM32F4XX boards or if you need WiFi ESP are also low cost.

**Reply by RafaelxVS** [reply]
> STM32F4XX boards or if you need WiFi ESP are also low cost.

Is this a good option? [Link](https://pt.aliexpress.com/item/4001049840288.html?spm=a2g0o.productlist.main.25.4aedc3d6JiBov3&algo_pvid=493cb0d2-5a7a-4721-86c9-10c3ff925185&algo_exp_id=493cb0d2-5a7a-4721-86c9-10c3ff925185-12&pdp_ext_f=%7B%22sku_id%22%3A%2210000014308323788%22%7D&pdp_npi=2%40dis%21BRL%2118.98%2115.59%21%21%21%21%21%402102172f16754524193821736d0674%2110000014308323788%21sea&curPageLogUid=VQVntYxfU6ku)

I'm in Brazil, Aliexpress is the easiest place to buy.

**Reply by Bodmer** [reply]
I would go for the slightly more expensive stm32f411 as it has more RAM. Bear in mind that you need a programming adapter with these simple STM32 boards.

An [RP2040 Pico](https://pt.aliexpress.com/item/1005004546639410.html?spm=a2g0o.productlist.main.1.2314MnFaMnFaXj&algo_pvid=fbb7c917-0474-4338-878d-68041ab100df&algo_exp_id=fbb7c917-0474-4338-878d-68041ab100df-0&pdp_ext_f=%7B%22sku_id%22%3A%2212000029562074385%22%7D&pdp_npi=2%40dis%21GBP%216.37%214.46%21%21%21%21%21%402100bb6416754573170426168d073e%2112000029562074385%21sea&curPageLogUid=FBPhxPE6AlM8) would be a better choice.

**Reply by Bodmer** [reply]
There seem to be some cheaper [RP2040 clone boards](https://pt.aliexpress.com/item/1005003970089918.html?spm=a2g0o.detail.1000060.1.48c4411dkuBCa2&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=83083c60-f569-4ce1-986a-8575b5242b76&_t=gps-id%3ApcDetailBottomMoreThisSeller%2Cscm-url%3A1007.13339.291025.0%2Cpvid%3A83083c60-f569-4ce1-986a-8575b5242b76%2Ctpp_buckets%3A668%232846%238112%231997&pdp_ext_f=%7B%22sku_id%22%3A%2212000029017753273%22%2C%22sceneId%22%3A%223339%22%7D&pdp_npi=2%40dis%21GBP%212.54%212.08%21%21%21%21%21%400b0a23a916754574402383465e991b%2112000029017753273%21rec&gatewayAdapt=glo2bra) available but I have not tried these.
